flake.nix

```diff
@@ -34,6 +38,19 @@
           self.packages.${system}.python
           self.packages.${system}.ros
         ];
+        # Precompute the BASE_DIR path
+        baseDir = pkgs.python3Packages.example-robot-data.outPath;
+        # Define the shared shell hook, referencing the precomputed path
+        sharedShellHook = ''
+          if [ -z "$BASE_DIR" ]; then
+            echo "Error: Could not locate the example-robot-data package." >&2
+          else
+            SHARE_DIR=$BASE_DIR
+            export AMENT_PREFIX_PATH=$SHARE_DIR:$AMENT_PREFIX_PATH
+            export ROS_PACKAGE_PATH=$SHARE_DIR/share:$ROS_PACKAGE_PATH
+            echo "Added $SHARE_DIR to AMENT_PREFIX_PATH and ROS_PACKAGE_PATH"
+          fi
+        '';
```

Comment:
If I understand correctly, you're doing all of that to get example-robot-data in AMENT_PREFIX_PATH.
Maybe we could just add that package to the ros buildEnv ?

Reply:
This is a trial. The idea is that the tutorial hpp do no run in this env because it can't find example robot data stuff upon xacro.
xacro is ran and fail to find some example-robot-data paths. I assume the newest version of xacro would look into AMENT_PACKAGE_PATH or ROS_PACKAGE_PATH. So before asking to modify example-robot-data I tried this and this fails for now. I need to talk to Florent about this.

Reply:
which can be done easily thanks to https://github.com/jrl-umi3218/jrl-cmakemodules/pull/717

Reply:
Actually, no need for that: python packages are propagated too : https://github.com/lopsided98/nix-ros-overlay/blob/master/distros/build-env/default.nix#L21

Reply:
Please test #12 :)